pkg/cmd/util/tokencmd/request_token.go

```diff
@@ -92,8 +93,27 @@ func (o *RequestTokenOptions) RequestToken() (string, error) {
 		return "", err
 	}
 
+	// TODO get from discovery endpoint?
+	config := &osincli.ClientConfig{
+		ClientId:                 o.ClientID,
+		AuthorizeUrl:             util.OpenShiftOAuthAuthorizeURL(o.ClientConfig.Host),
+		TokenUrl:                 util.OpenShiftOAuthTokenURL(o.ClientConfig.Host),
+		RedirectUrl:              util.OpenShiftOAuthTokenImplicitURL(o.ClientConfig.Host),
+		SendClientSecretInParams: true, // we have no secret, just a client id
```

Comment:
should this just be false, then, or are we anticipating future use of a secret?

Reply:
We need to set this to true b/c we want to send the client ID in the params.  It will send the empty string as the secret.  There cannot be any future use of a secret since its impossible to protect (would have to be embedded inside `oc`).

Reply:
params vs basic header is orthogonal to use of a secret or not